### DISCUSSION TITLE
wol for local server

### DISCUSSION THREAD

**Discussion by umtauscher** [music-assistant/support#3052] [Q&A] [answered]
I just discovered Musicassistant yesterday and I am very impressed with in in my homeassistant installation.
My music is shared from a NAS which is in standby most of the time.
When I use it with Kodi, the kodi installation automatically sends a wol package to wake up the server.

Is there any way I could accomplish a similar automation with Musicassistant when a new song is started?
TIA

**Comment by marcelveldt** [comment]
You are using it with Home Assistant so your possibilities are endless. So I would create an automation in HA. For example if you turn on a player or something along those lines.

**Reply by umtauscher** [reply]
A bit more specific would really help.
What trigger do you suggest?

**Comment by umtauscher** [comment]
Ok, I mangaged to send wol on any change on one of the speakers.
This is the only way it worked, because when I set it on „Start playing“ it never gets triggered, because MA fails before the playing starts.
Additionally the automation also gets triggered, when I play Radio, which doesn‘t involve the server.
I think this is a workarround at best. 
Since there don‘t seem to be usable triggers form the server.
The best solution would be, if I could add a wol device to a remote network share.

**Comment by umtauscher** [comment]
Sorry, I have to correct my previous message.
This is solution is unusable for me. The server constantly wakes up, even when there are other sources  not MA are using the players.
I have used the MA player entities in the automation, but it also gets triggered, when I use the musiccast entities that were there before.

**Reply by OzGav** [reply, marked as answer]
At this time MA does not have the functionality you mention Kodi has. If you cannot identify a specific entity state that you can automate off then what you want to do may not be possible. You are welcome to open a feature request for this and see how many votes you get from others.

**Reply by OzGav** [reply]
With MA2.4.0b23 and later there is now functionality to map a power control to a HA entity. You may be able to use this to map a switch controlling the WOL functionality to a MA player power button?

**Reply by umtauscher** [reply]
I thought it could not be that difficult to send a wol packet to the nas before accessing a share and wait a
little while before throwing a error message.

Sending a wol packet by hand/button before starting MA is far too  unsophisticated  for my taste.
